src/pages/api/auth/[...nextauth].js

```diff
@@ -1,28 +1,48 @@
-import NextAuth from "next-auth/next"
-import GoogleProvider from "next-auth/providers/google"
-import EmailProvider from "next-auth/providers/email"
-import { SupabaseAdapter } from "@next-auth/supabase-adapter"
+import NextAuth from 'next-auth/next'
+import GoogleProvider from 'next-auth/providers/google'
+import EmailProvider from 'next-auth/providers/email'
+import { SupabaseAdapter } from '@next-auth/supabase-adapter'
+import jwt from 'jsonwebtoken'
+
 export const authOptions = {
   providers: [
-    // GoogleProvider({
-    //   clientId: process.env.GOOGLE_ID,
-    //   clientSecret: process.env.GOOGLE_SECRET,
-    // }),
+    GoogleProvider({
+      clientId: process.env.GOOGLE_ID,
+      clientSecret: process.env.GOOGLE_SECRET
+    }),
     EmailProvider({
       server: {
         host: process.env.EMAIL_SERVER_HOST,
         port: process.env.EMAIL_SERVER_PORT,
         auth: {
           user: process.env.EMAIL_SERVER_USER,
-          pass: process.env.EMAIL_SERVER_PASSWORD,
-        },
+          pass: process.env.EMAIL_SERVER_PASSWORD
+        }
       },
-      from: process.env.EMAIL_FROM,
-    }),
+      from: process.env.EMAIL_FROM
+    })
   ],
   adapter: SupabaseAdapter({
     url: process.env.NEXT_PUBLIC_SUPABASE_URL,
-    secret: process.env.SUPABASE_SERVICE_ROLE_KEY,
+    secret: process.env.SUPABASE_SERVICE_ROLE_KEY
   }),
+  callbacks: {
+    async session({ session, user }) {
```

Comment:
I need to look into this further, but I think we might be able to set the role here.

Reply:
**Suggestion:**
@JIMMY-YA0, what are you thoughts on looking up the user in Supabase here, and inserting the role here?